src/polyhash/index.js

```diff
@@ -79,39 +79,122 @@ const decode = [
 ];
 
 /**
- * Converts an array of points into a polyhash, locationId-polygon
+ * Converts a geometry into a polyhash, locationId-polygon
  *
- * @param {point[]} points
- * @param {int} locationIdPrecision
+ * @param {*} input, coordindates, locationIds, polygon or stringId
+ * @param {*} locationIdPrecision
  */
-function toPolyhash(points, locationIdPrecision = 9) {
+function toPolyhash(input, locationIdPrecision = 9) {
   if (locationIdPrecision > maxLocationIdPrecision) {
     console.error(`Invalid locationId precision ${locationIdPrecision}. Maximum supported is ${maxLocationIdPrecision}`)
     return null
   }
 
-  const polyhash = points.map(point => unl.encode(point[0], point[1], locationIdPrecision))
-  return deflate(polyhash)
+  let result = null;
+  if (typeof input === 'string' || input instanceof String) {
+    result = input;
+  }
+  else {
+    const locationIds = toLocationIds(input)
+    result = compress(locationIds)
+  }
+  return result
 }
 
 /**
- * Returns an array of coordinates, the polygon represented by the given polyhash
+ * Convert a geometry into a list of coordinates
  *
- * @param {*} polyhash
+ * @param {*} input, coordindates, locationIds, polygon or stringId
+ * @param {*} locationIdPrecision
  */
-function toCoordinates(polyhash) {
-  return inflate(polyhash).map(locationId => {
-    const point = unl.decode(locationId);
-    return [parseFloat(point.lon.toFixed(6)), parseFloat(point.lat.toFixed(6))]
-  });
+function toCoordinates(input) {
+  try {
+    let result = [];
+    if (input.features) {
+      if (input.features.length) {
+        result.push(...
+          input.features[0].geometry.coordinates.flatMap((arr) =>
+            arr.map((coords) => [coords[0], coords[1]])
+          )
+        );
+      }
+    } else if (input.geometry && input.geometry.type === "MultiPolygon") {
+      const largestPolygon = input.geometry.coordinates.forEach((coords) => {
+        result.push(...coords);
+      });
+    } else if (input.geometry && input.geometry.type === "Polygon") {
+      result.push(...input.geometry.coordinates);
+    }
+    // Array of coords or lists
+    else if (Array.isArray(input) && input.length) {
+      // array of coords
+      if (Array.isArray(input[0])) {
+        result = input;
+      }
+      // Array of locationId
+      else {
+        result = input.map((locationId) => {
+          const point = unl.decode(locationId);
+          return [
+            parseFloat(point.lon.toFixed(6)),
+            parseFloat(point.lat.toFixed(6)),
+          ];
+        });
+      }
+    }
+    else {
+      // compressed string Id
+      const locationIs = decompress(input)
+      result = toCoordinates(locationIs)      
+    }
+    // polyhash
+    return result;
+  } catch (err) {
+    console.error(
+      `Failed to convert ${JSON.stringify(
+        input
+      )} to coordinates, reason: ${err}`
```

Comment:
I think it's probably best to hide some errors from our users. We can catch them and return our own error messages for some possible cases

Reply:
Done